Source/Diagnostics/Diagnostics.H

```diff
@@ -18,99 +18,125 @@ class Diagnostics
 {
 public:
     Diagnostics (int i, std::string name);
-    ~Diagnostics ();
+    /** Virtual Destructor to handle clean destruction of derived classes */
+    virtual ~Diagnostics () ;
     /** Pack (stack) all fields in the cell-centered output MultiFab m_mf_output.
-     * Fields are computed (e.g., cell-centered) on-the-fly using a functor.*/
+     * Fields are computed (e.g., cell-centered or back-transformed)
+       on-the-fly using a functor.*/
```

Comment:
You can separate title from description in doxygen syntax like this:
```suggestion
     * 
     * Fields are computed (e.g., cell-centered or back-transformed)
       on-the-fly using a functor. */
```